src/lib-imap-client/imapc-connection.c

```diff
@@ -382,6 +382,10 @@ static void imapc_connection_set_state(struct imapc_connection *conn,
 
 		conn->selecting_box = NULL;
 		conn->selected_box = NULL;
+
+		if (conn->ips != NULL)
```

Comment:
..and it's used all over the code everywhere, so better to be consistent here also and not have the if-check :)